### PR TITLE
feat: mintable with permit

### LINKED ISSUE #11880: OptimismMintableERC20: Add permit support + native permit2 support
Add `permit` support to the `OptimismMintableERC20` token implementation. This can easily be done using the `ERC20Permit` contract in openzeppelin. Also add native support for `permit2` by enforcing that the `allowance` for `permit2` always returns `type(uint256).max`.

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Adds `permit` support to the `OptimismMintableERC20Token` as well as
adding native support for `permit2` by default.

Closes https://github.com/ethereum-optimism/optimism/issues/11880

### CODE REVIEW

**Comment by tynes** [comment]
I kind of want to drop oz here in favor of solady. It would be a straight up scalability improvement

File: packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol [hunk 1]
```diff
@@ -54,12 +64,35 @@ contract OptimismMintableERC20 is IOptimismMintableERC20, ILegacyMintableERC20,
...
+        return DECIMALS;
+    }
+
+    /// @notice Returns the allowance for a spender on the owner's tokens.
+    ///         If the spender is the permit2 address, returns the maximum uint256 value.
+    /// @param _owner   owner of the tokens.
+    /// @param _spender spender of the tokens.
+    /// @return Allowance for the spender.
+    function allowance(address _owner, address _spender) public view override returns (uint256) {
+        if (_spender == PERMIT2()) {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol:90]: Hmmm why not use the constant directly?

Comment (smartcontracts) [reply packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol:90]: Like: `if (_spender == Preinstalls.Permit2) {...}`

Comment (tynes) [reply packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol:90]: Contracts generally expose it as a public getter

Comment (smartcontracts) [reply packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol:90]: Oh I didn't realize that

**Comment by mds1** [comment]
> I kind of want to drop oz here in favor of solady. It would be a straight up scalability improvement

Discussed with @tynes and we decided to leave this out of scope for this PR. We want to merge this soon, and switching to solady should be done deliberately with some thought, because:
- It changes the ABI by removing `increaseAllowance` and `decreaseAllowance` methods. We're ok with removing these methods, but want to make sure us and users have time to consider effects of this breaking change
- We need to validate it's security by (1) making sure the solady ERC20 was in scope for its audit, and/or (2) finding a commit that's been widely used in production

**Review by mds1 — APPROVED** [review]
Reviewed synchronously with @tynes, this LGTM

**Pull request merged**